include/boost/geometry/formulas/differential_quantities.hpp

```diff
@@ -0,0 +1,269 @@
+        CT const sin3_2sig2 = math::sqr(sin_2sig2) * sin_2sig2;
+        CT const sin3_2sig_12 = sin3_2sig2 - sin3_2sig1;
+
+        CT const A = (9 * cos_alp0_quad - 12 * cos_alp0_sqr) * sin_4sig_12;
+        CT const B = 4 * cos_alp0_quad * sin3_2sig_12;
+        CT const C = (-48 * cos_alp0_quad + 96 * cos_alp0_sqr - 64) * sin_2sig_12;
+        CT const D = (60 * cos_alp0_quad - 144 * cos_alp0_sqr + 128) * sig_12;
+
+        CT const L3 = (A + B + C + D) / 64;
+
+        // Order 3 and higher
+        return cos_alp0_sqr * f * (L1 + f * (L2 + f * L3));
+    }
+
+    /*! Approximation of J12, expanded into taylor series in e'^2
+        Maxima script:
+        k2: ca02 * ep2;
+        assume(sig > 0);
+        I1(sig):= integrate(sqrt(1 + k2 * sin(s)^2), s, 0, sig);
+        I2(sig):= integrate(1/sqrt(1 + k2 * sin(s)^2), s, 0, sig);
+        J(sig):= I1(sig) - I2(sig);
+        S: taylor(J(sig), ep2, 0, 3);
+        S1: factor( integrate(sin(s)^2,s,0,sig)*ca02*ep2 );
+        S2: factor( (integrate(sin(s)^4,s,0,sig)*ca02^2*ep2^2)/2 );
+        S3: factor( (3*integrate(sin(s)^6,s,0,sig)*ca02^3*ep2^3)/8 );
+    */
+    static inline CT J12_ep_sqr(CT const& sin_sig1, CT const& cos_sig1,
+                                CT const& sin_sig2, CT const& cos_sig2,
+                                CT const& cos_alp0_sqr, CT const& ep_sqr)
+    {
+        if (Order == 0)
+        {
+            return 0;
+        }
+
+        CT const c2a0ep2 = cos_alp0_sqr * ep_sqr;
+
+        CT const sig_12 = atan2(cos_sig1 * sin_sig2 - sin_sig1 * cos_sig2,
+                                cos_sig1 * cos_sig2 + sin_sig1 * sin_sig2); // sig2 - sig1
+        CT const sin_2sig1 = 2 * cos_sig1 * sin_sig1; // sin(2sig1)
+        CT const sin_2sig2 = 2 * cos_sig2 * sin_sig2; // sin(2sig2)
+        CT const sin_2sig_12 = sin_2sig2 - sin_2sig1;
+
+        CT const L1 = (2 * sig_12 - sin_2sig_12) / 4;
+
+        if (Order == 1)
+        {
+            return c2a0ep2 * L1;
+        }
+        
+        CT const sin_4sig1 = 2 * sin_2sig1 * (math::sqr(cos_sig1) - math::sqr(sin_sig1)); // sin(4sig1)
+        CT const sin_4sig2 = 2 * sin_2sig2 * (math::sqr(cos_sig2) - math::sqr(sin_sig2)); // sin(4sig2)
+        CT const sin_4sig_12 = sin_4sig2 - sin_4sig1;
+        
+        CT const L2 = (sin_4sig_12 - 8 * sin_2sig_12 + 12 * sig_12) / 64;
+
+        if (Order == 2)
+        {
+            return c2a0ep2 * (L1 + c2a0ep2 * L2);
+        }
+
+        CT const sin3_2sig1 = math::sqr(sin_2sig1) * sin_2sig1;
+        CT const sin3_2sig2 = math::sqr(sin_2sig2) * sin_2sig2;
+        CT const sin3_2sig_12 = sin3_2sig2 - sin3_2sig1;
+
+        CT const L3 = (9 * sin_4sig_12 + 4 * sin3_2sig_12 - 48 * sin_2sig_12 + 60 * sig_12) / 512;
+
+        // Order 3 and higher
+        return c2a0ep2 * (L1 + c2a0ep2 * (L2 + c2a0ep2 * L3));
+    }
+
+    static inline void normalize(CT & x, CT & y)
+    {
+        CT const l = math::sqrt(math::sqr(x) + math::sqr(y));
```

Comment:
Please change l into len, because l/1 are difficult to distinguish - and we divide by it, how sure are we that it is non zero?